src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -66,8 +66,12 @@ public static bool IsStructMarshallingRequired(TypeDesc typeDesc)
                 typeDesc = typeDesc.GetParameterType();
             }
 
-            if (typeDesc.Category != TypeFlags.ValueType)
+            typeDesc = typeDesc.UnderlyingType;
+
+            if (typeDesc.IsPrimitive)
```

Comment:
There are primitive types which require marshalling, such as bool, char.

Reply:
This function does not quite do what the name suggests. It used to return `false` for all primitive types (including bool and char) even before my change. I have added TODO for this because of fixing this does not seem to be easy.